lib/core/security/tokenRepository.ts

```diff
@@ -308,36 +338,41 @@ export class TokenRepository extends ObjectRepository<Token> {
       return this.anonymousToken;
     }
 
+    const isApiKey = token.startsWith(Token.APIKEY_PREFIX);
+    const tokenWithoutPrefix = this.removeTokenPrefix(token);
+
     let decoded = null;
 
     try {
-      decoded = jwt.verify(this.removeTokenPrefix(token), global.kuzzle.secret);
-
+      decoded = jwt.verify(tokenWithoutPrefix, global.kuzzle.secret);
       // probably forged token => throw without providing any information
       if (!decoded._id) {
         throw new jwt.JsonWebTokenError("Invalid token");
       }
     } catch (err) {
-      if (err instanceof jwt.TokenExpiredError) {
-        throw securityError.get("expired");
-      }
-
       if (err instanceof jwt.JsonWebTokenError) {
         throw securityError.get("invalid");
       }
 
+      if (err instanceof jwt.TokenExpiredError) {
+        throw securityError.get("expired");
+      }
+
       throw securityError.getFrom(err, "verification_error", err.message);
     }
 
-    let userToken: Token;
+    if (isApiKey) {
+      return this._verifyApiKey(decoded, token);
+    }
+
+    let userToken;
```

Comment:
After discussing with @Juiced66. This function should be reworked to give two different path.

1. Path one, will only use the _verifyApiKey(token) and decode the token the way it need to be from this function
1. Path two, add a new function _verifyAuthToken(token) that also check the verification the way it need.

We may need to have a common jwt.verify function as well.

This will allow us to have two distinct ways of testing the validity of theses two way of logging in.

Reply:
As discussed this mornig, those changes will be done in another PR as this one is working as intended